pycbc/waveform/ringdown.py

```diff
@@ -76,22 +76,24 @@ def format_lmns(lmns):
     given, raise an error.
     """
 
-    # Case 1: the list is in a string "[221, 331]"
-    if isinstance(lmns, (str, unicode)):
-        # strip off brackets
-        lmns = lmns.strip('[]')
-        # convert to list
-        lmns = lmns.split(',')
+    # Case 1: the list is in a string "['221', '331']"
+    # In Python3 this might be "[b'221', b'331']"
+    if isinstance(lmns, str):
+        # strip off brackets and convert to list
+        lmns = lmns.strip('[]').split(',')
 
     # Case 2: a list with only one string with a list ["221', '331"]
-    elif (len(lmns) == 1 and isinstance(lmns[0], (str, unicode))
+    # In Python3 this might be ["b221', b'331"]
+    elif (len(lmns) == 1 and isinstance(lmns[0], str)
           and len(lmns[0]) > 3):
         lmns = lmns[0].split(',')
 
     out = []
     # Cycle over the lmns to ensure that we get back a list of strings that
     # are three digits long, and that nmodes!=0
     for lmn in lmns:
+        # For Python3
+        lmn = lmn.strip(" b'")
```

Comment:
@cdcapano I have added this line for python3: when reading the lmns dataset from an hdf file one will get `array([b'221', b'331'], dtype='|S3')`, which will be converted to a string by the workflow hence becoming `"[b'221', b'331']"`. Since the bytes object indicator `b` is now inside the string, it is not recognized anymore and `.decode()` will not work (furthermore, the next line will fail when trying to do `int("b'221'")`. Here I am getting rid of the `b` that is in the string (and extra spaces or quotes).
If anyone knows a better solution (@spxiwh @ahnitz ?), please let me know. Otherwise, if the tests do not fail, this should be ready to merge.

Reply:
Where does the `array([b'221', b'331'], dtype='|S3')` get cast to `str`. Fixing this there would be better

Reply:
I guess this is an example for how I'm trying to handle this:
https://github.com/gwastro/pycbc/pull/2774

Reply:
@spxiwh An example is in PR #2738 (I have put a comment for you there showing the line in the code). There is also another piece of code related to this 'lmns' parameter in https://github.com/gwastro/pycbc/blob/master/pycbc/distributions/__init__.py#L95 that uses `str`. Do I need to make some change there too for python3?

Reply:
I don't see any particular issue in #2738, or in https://github.com/gwastro/pycbc/blob/master/pycbc/distributions/__init__.py#L95 .. Although I think it's strange to cast this to `str` rather than preserving the input type.

I think that some of the underlying problems here likely come from the HDF injection files (and the `_HDFInjectionSet` class and `inject.py`). I think there's a number of python3 issues in that code, which will need fixing, but I'd fix it there rather than trying to add multiple patches in places like this one. (If you're not planning on running with python3 any time soon, this can go on the to-do pile, but it will need doing at some point).

Reply:
I can comment out this line for now and add a FIXME note to the python3 comment for the future.
Another option that I suggested to @cdcapano would be to change the expected format of the lmns to be a list of ints instead of a list of strings. Then the hdf dataset would not have the `b` anymore, and we could just convert the ints to strings in the ringdown module when we need to read them. But that would be on a separate PR, as I am not sure right now if that would require more changes than just the ringdown module and the https://github.com/gwastro/pycbc/blob/master/pycbc/distributions/__init__.py#L95 (and I haven't heard from Colin on this yet, we might need to discuss it a bit further)

Reply:
This can definitely work when using strings in the HDF files, we just need to be more careful about identifying what the datatypes actually are. But first we would need a test script/executable/workflow that you want to get running in python3 where this code is used.

Reply:
Ok, I can set that up. Any instructions on how to do that or example to look at?